loadtest/control/simulcontroller/actions.go

```diff
@@ -1831,3 +1832,152 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
+			// permalinks which emulates organic behavior.
+			postID = p.Message[index+4 : index+30]
+		}
+	}
+	if postID == "" {
+		return control.UserActionResponse{Info: "no permalink found"}
+	}
+
+	// We fetch the post thread.
+	postIds, _, err := u.GetPostThreadWithOpts(postID, "", model.GetPostsOptions{
+		CollapsedThreads: true,
+		Direction:        "down",
+		PerPage:          25,
+	})
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	oldestPost, err := u.Store().Post(postIds[0])
+	if err != nil && !errors.Is(err, memstore.ErrPostNotFound) {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	ch, err := u.Store().Channel(oldestPost.ChannelId)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	// We fetch the channel if it's not present in state.
+	// This is similar to what the webapp does.
+	if ch == nil {
+		if err := u.GetChannel(oldestPost.ChannelId); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		// We fetch the channel data from store again.
+		ch, err = u.Store().Channel(oldestPost.ChannelId)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	// We check if membership data is already there. If not,
+	// we call getChannelMember, and add the user to the channel
+	// if it's not a private channel.
+	cm, err := u.Store().ChannelMember(ch.Id, u.Store().Id())
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	if cm.UserId == "" {
+		err := u.GetChannelMember(ch.Id, u.Store().Id())
+		if err != nil {
+			// Resorting to string matching, because the API doesn't return the
+			// actual Response object.
+			if strings.Contains(err.Error(), "You do not have the appropriate permissions") {
+				if ch.Type != model.ChannelTypePrivate {
+					if err := u.AddChannelMember(ch.Id, u.Store().Id()); err != nil {
+						return control.UserActionResponse{Err: control.NewUserError(err)}
+					}
+				}
+			} else {
+				return control.UserActionResponse{Err: control.NewUserError(err)}
+			}
+		}
+	}
+
+	currentTeam, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if currentTeam == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	if _, err := u.GetChannelsForTeamForUser(currentTeam.Id, u.Store().Id(), true); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if err := u.GetChannelMembersForUser(u.Store().Id(), currentTeam.Id); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// Switch to channel.
+	if resp := viewChannel(u, ch); resp.Err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(resp.Err)}
+	}
```

Comment:
Not something to change in this PR, this is something I realized in the frequencies analysis, but worth pointing out:
We call the `viewChannel` helper action in many places, so we need to fix the frequency of the `switchChannel` action (which is based on the `viewChannel` handler, of course).

Reply:
Right .. it's tricky.